test/pykernel/gen/colman_fused_eltwise_bcast.py

```diff
@@ -45,13 +48,28 @@ async def compute():
         
         for ct in range(start_col_tile, end_col_tile):
             # Reuse C across rows of A, B
-            c_block = c_in_cb.wait().pop()
+            # The following waits until *shape* number of tiles (1 in this case) is available and returns a RingView object as defined in:
+            # https://github.com/tenstorrent/tt-lang/blob/main/python/sim/cbsim/ringview.py.
+            # TODO: Perhaps consider making RingView pointers that come from wait()/reserve() read/write only respectively?
+            c_block = c_in_cb.wait() 
             for rt_block in range(row_tiles // granularity):
-                a_block = a_in_cb.wait().pop()
-                b_block = b_in_cb.wait().pop()
+                # again, these wait until *shape* number of tiles (*granularity*, i.e 4 in this case) are available and return RingView pointers:
+                a_block = a_in_cb.wait() # blocking 
+                b_block = b_in_cb.wait() # blocking
                 # NOTE: Please consider making non-approx the default for eltwise unary, but leave the option for the user to specify approx=True
+                out_block = out_cb.reserve() # blocking
+                # RingView operations, these also need to be defined
                 out_block = a_block * b_block + tt.tanh(c_block, approx=False)
-                out_cb.push(out_block)
+                # finalize push, this advances the cb pointers, the writing happened at the line above
+                out_cb.push()
```

Comment:
Push also doesn't exist; this would also be a pop to signal data is ready

Reply:
Pop advances the cb counter, and effectively invalidates `out_block`, I don't think that's how data-ready signaling is accomplished. The metal interface has push specifically for that purpose, and I don't think it's appropriate to use pop in this case -- a different thread should be popping this CB entry (e.g., dm).

Another thing I don't quite understand is why this has to be explicit. The compiler can automatically insert push after all operations involved in the assignment are completed. But I guess if we want users to control when signaling happens, then you need push (or something equivalent, but not pop, IMO).